docs/current_docs/manuals/user/remotes/remote-repositories.mdx

```diff
@@ -11,3 +11,91 @@ Here is an example of a Go builder Dagger Function that accepts a remote Git add
+
+### Supported Protocols and Schemes
+
+| Protocol | Scheme            | Private Modules Support | Example                                      |
+|----------|-------------------|-------------------------|----------------------------------------------|
+| HTTP     | Go-like ref style | In progress               | `github.com/username/repo[/subdir][@version]`  |
+| HTTP     | Git HTTP style    | In progress               | `https://github.com/username/repo.git[/subdir][@version]` |
+| SSH      | SCP-like          | Supported               | `git@github.com:username/repo.git[/subdir][@version]`     |
+| SSH      | Explicit SSH      | Supported               | `ssh://git@github.com/username/repo.git[/subdir][@version]` |
+
+### Additional Ref Options
+
+Dagger provides flexibility in referencing modules through the following options:
+
+- The `.git` extension is optional for HTTP refs or explicit SSH refs, except for [GitLab, when referencing modules stored on a private repo or private subgroup](https://gitlab.com/gitlab-org/gitlab-foss/-/blob/master/lib/gitlab/middleware/go.rb#L229-237).
+- **Monorepo support**: Append `/subpath` to access a specific subdirectory within the repository.
+- **Version specification**: Add `@version` to target a particular version of the module. This can be a tag, branch name, or full commit hash. If omitted, the default branch is used.
+
+These options are optional.
+
+### SSH Refs and Authentication
+
+Here is an overview of how Dagger handles authentication when using SSH refs:
+
+Dagger mounts the socket specified by your host's `SSH_AUTH_SOCK` environment variable to the engine. This is essential for SSH refs, as most Git servers use your SSH key for authentication and tracking purposes, even when cloning public repositories. Ensure that the `SSH_AUTH_SOCK` environment variable is properly set in your environment when using SSH refs.
+
+For detailed instructions on setting up SSH authentication, including how to generate SSH keys, start the SSH agent, and add your keys, please refer to the following external documentation:
+
+[Connecting to GitHub with SSH](https://docs.github.com/en/authentication/connecting-to-github-with-ssh)
+
+This guide provides comprehensive information on:
+
+- Generating SSH key pairs
+- Setting up public key authentication on Linux, macOS, and Windows
+- Configuring SSH clients
+- Best practices for SSH key management
+
+You can ensure that your SSH authentication is properly configured for use with Dagger by following this guide.
+
+### Best Practices
+
+For quick and easy referencing:
+- Copy the repository ref from your preferred Git server's UI.
+- Add a `/subpath` if you need to target a specific directory within the repository.
+- Append `@version` if you want to specify a particular version or commit.
+
+### Known Limitations and Workarounds
```

Comment:
WDT of moving the content in this section to the troubleshooting page and instead adding a notification box here like: This feature is still under development. If something doesn't work as expected, check the [list of known issues and troubleshooting steps], or open a GitHub issue. We can link to https://docs.dagger.io/manuals/user/troubleshooting

Reply:
I personally prefer if we leave this here. The troubleshooting page has a different purpose IMO. In this particular case, we're aware about some limitations in the current implementation and it's best if the user can read about those here while following this page instead of finding them later in the troubleshooting section.